Makefile

```diff
@@ -1,133 +1,24 @@
-ANTLR=java -cp $(ANTLR_DEPS) org.antlr.v4.Tool
-ANTLR_SRC=src/main/antlr4/com/zendesk/maxwell/schema/ddl/mysql.g4
-ANTLR_IMPORTS=src/main/antlr4/imports
-ANTLR_DIR=target/generated-sources/src/main/antlr4/com/zendesk/maxwell/schema/ddl
-ANTLR_OUTPUT=$(ANTLR_DIR)/mysqlBaseListener.java $(ANTLR_DIR)/mysqlLexer.java $(ANTLR_DIR)/mysqlListener.java $(ANTLR_DIR)/mysqlParser.java
-
-$(ANTLR_IMPORTS)/mysql_literal_tokens.g4: $(ANTLR_IMPORTS)/generate_tokens.rb
-	ruby $(ANTLR_IMPORTS)/generate_tokens.rb
-
-$(ANTLR_OUTPUT): $(ANTLR_SRC) $(ANTLR_IMPORTS)/*.g4
-	${ANTLR} -package com.zendesk.maxwell.schema.ddl -lib $(ANTLR_IMPORTS) -o target/generated-sources $(ANTLR_SRC)
-
-compile-antlr: $(ANTLR_OUTPUT)
-
-JAVA_SOURCE = $(shell find src/main/java -name '*.java')
-JAVA_DEPENDS = $(shell build/maven_fetcher -p -o target/dependency $(LOCK_KAFKA_VERSION))
-
-target/.java: $(ANTLR_OUTPUT) $(JAVA_SOURCE)
-	@mkdir -p target/classes
-	@# Fetch jar so we can run it locally (bin/maxwell)
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_08) --skip-dependencies -o target/dependency >/dev/null
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_010) --skip-dependencies -o target/dependency >/dev/null
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_01010) --skip-dependencies -o target/dependency >/dev/null
-	build/maven_fetcher -f org.apache.kafka/kafka-clients/$(ADDITIONAL_PACKAGED_KAFKA_01021) --skip-dependencies -o target/dependency >/dev/null
-	$(JAVAC) -classpath $(JAVA_DEPENDS) $(JAVAC_FLAGS) $?
-	@touch target/.java
-
-copy-resources:
-	@cp -a src/main/resources/* target/classes
+KAFKA_VERSION ?= 0.9.0.1
+KAFKA_PROFILE = kafka-${KAFKA_VERSION}
 
-compile-java: target/.java
-compile: compile-antlr compile-java copy-resources
-
-
-
-JAVA_TEST_DEPENDS = $(shell build/maven_fetcher -p -o target/dependency-test -s test $(LOCK_KAFKA_VERSION))
-JAVA_TEST_SOURCE=$(shell find src/test/java -name '*.java')
-target/.java-test: $(JAVA_TEST_SOURCE)
-	@mkdir -p target/test-classes
-	cp -a src/test/resources/* target/test-classes
-	javac -d target/test-classes -sourcepath src/main/java:src/test/java:target/generated-sources -classpath target/classes:$(JAVA_TEST_DEPENDS) \
-		-g -target 1.8 -source 1.8 -encoding UTF-8 $?
-	@touch target/.java-test
-
-compile-test: compile target/.java-test
-
-TEST_CLASSES=$(shell build/get-test-classes)
-
-define get_test_parameters
-	$(eval filter=$(shell echo $(1) | perl -ne '/#(.*)/; print "--filter=com.zendesk.maxwell.JUnitNameFilterFactory=$$1" if $$1')) \
-	$(eval class_pattern=$(shell echo $(1) | perl -ne '/test\.(\w+)/; print $$1 if $$1')) \
-	$(eval classes=$(if $(class_pattern),$(filter %$(class_pattern),$(TEST_CLASSES)),$(TEST_CLASSES))) \
-	$(filter) $(classes)
-endef
-
-JAVA_TEST_PARAMS=-Djava.util.logging.config.file=./src/test/resources/logging.properties
+all: compile
```

Comment:
Does this include antlr compilation?

Reply:
Yep - I did a full (git) clean, and compile still works. Looks like we already have the antlr maven plugin, we were just replicating it in the Makefile...

> [INFO] --- antlr4-maven-plugin:4.5:antlr4 (antlr) @ maxwell ---
[INFO] ANTLR 4: Processing source directory /Users/tcuthbertson/Code/zendesk/maxwell/src/main/antlr4
[INFO] Processing grammar: com/zendesk/maxwell/schema/ddl/mysql.g4

Reply:
👍